src/main/java/org/apache/maven/shared/jar/classes/JarClassesAnalysis.java

```diff
@@ -154,7 +154,7 @@ private JarClasses analyzeMultiRelease(JarAnalyzer jarAnalyzer) {
         for (Map.Entry<Integer, JarVersionedRuntime> runtimeVersionEntry : runtimeVersionsMap.entrySet()) {
             Integer version = runtimeVersionEntry.getKey();
             String jdkRevision = runtimeVersionEntry.getValue().getJarClasses().getJdkRevision();
-            if (!version.equals(Integer.valueOf(jdkRevision))) {
+            if (jdkRevision != null && !version.toString().equals(jdkRevision)) {
                 logger.warn(
```

Comment:
Not this PR's question, but I wasn't here when it was introduced.
I don't understand why the warning.
https://docs.oracle.com/en/java/javase/17/docs/specs/jar/jar.html#multi-release-jar-files:
> A class file under a versioned directory, of version N say, in a multi-release JAR must have a class file version less than or equal to the class file version associated with Nth major version of a Java platform release.

Reply:
As I understand it, this say that you can put into a versioned directory any class that the JVM with version N is able to execute. However, when ALL classes are compiled targeting a lower JVM version, then you may wonder if this was actually what was intended. Since this library tries to analyze JAR files I feelt the warning was appropriate.

Reply:
> As I understand it, this say that you can put into a versioned directory any class that the JVM with version N is able to execute. However, when ALL classes are compiled targeting a lower JVM version, then you may wonder if this was actually what was intended. Since this library tries to analyze JAR files I feelt the warning was appropriate.

While I consider your understanding reasonable the spec says otherwise. Maybe with a newer Java version the impl can be different, but the code does not need to have a higher byte version.